### PR TITLE
btf: add String() for btfKind

### PR DESCRIPTION
**Pull request by florianl**
Signed-off-by: Lehner Florian <dev@der-flo.net>

This PR started by fixing a simple error message and resulted then in adding the missing `String()` for `btfKind`.

### CODE REVIEW

File: example_sock_elf_test.go [hunk 1]
```diff
@@ -77,8 +77,6 @@ var program = [...]byte{
 	0010, 0000, 0000, 0000, 0000, 0000, 0000, 0000, 0030, 0000, 0000, 0000, 0000, 0000, 0000, 0000,
 }
 
-const sockexPin = "/sys/fs/bpf/sockex1"
```
Comment (tklauser) [inline example_sock_elf_test.go]: This change looks unrelated. Maybe move into a separate commit?

File: internal/btf/btf_types.go [hunk 2]
```diff
@@ -61,6 +62,45 @@ type btfType struct {
...
+	case kindPointer:
+		return "Pointer"
+	case kindArray:
+		return "Array"
+	case kindStruct:
+		return "Struct"
+	case kindUnion:
+		return "Union"
+	case kindEnum:
+		return "Enum"
```
Comment (tklauser) [inline internal/btf/btf_types.go]: I'd use `Enumeration` here in order to stay consistent with the comment in `linux/btf.h`.
```suggestion
		return "Enumeration"
```

File: internal/btf/btf_types.go [hunk 3]
```diff
@@ -61,6 +62,45 @@ type btfType struct {
...
+	case kindTypedef:
+		return "Typedef"
+	case kindVolatile:
+		return "Volatile"
+	case kindConst:
+		return "Const"
+	case kindRestrict:
+		return "Restrict"
+	case kindFunc:
+		return "Func"
```
Comment (tklauser) [inline internal/btf/btf_types.go]: Same here, better stay consistent with `linux/btf.h`
```suggestion
		return "Function"
```

File: internal/btf/btf_types.go [hunk 4]
```diff
@@ -61,6 +62,45 @@ type btfType struct {
...
+	case kindFuncProto:
+		return "FuncProto"
```
Comment (tklauser) [inline internal/btf/btf_types.go]: Same here:
```suggestion
		return "Function Proto"
```

File: internal/btf/btf_types.go [hunk 5]
```diff
@@ -61,6 +62,45 @@ type btfType struct {
...
+	case kindVar:
+		return "Var"
```
Comment (tklauser) [inline internal/btf/btf_types.go]:
```suggestion
		return "Variable"
```

File: internal/btf/btf_types.go [hunk 6]
```diff
@@ -61,6 +62,45 @@ type btfType struct {
...
+	case kindDatasec:
+		return "Datasec"
```
Comment (tklauser) [inline internal/btf/btf_types.go]: `linux/btf.h` calls this just `Section`
```suggestion
		return "Section"
```

File: example_sock_elf_test.go (hunk 1 above)
Comment (florianl) [reply example_sock_elf_test.go]: I'v removed it from this PR.

**Comment by florianl** [comment]
I've applied the requested changes.

File: internal/btf/btf_types.go [hunk 7]
```diff
@@ -61,6 +62,45 @@ type btfType struct {
 	SizeType uint32
 }
 
+func (k btfKind) String() string {
+	switch k {
+	case kindUnknown:
+		return "Unknown"
+	case kindInt:
+		return "Int"
```
Comment (tklauser) [inline internal/btf/btf_types.go]: Sorry, I missed one in my previous review
```suggestion
		return "Integer"
```
